packages/web/src/pages/dashboard-page/components/ArtistContentSection.tsx

```diff
@@ -0,0 +1,461 @@
+import { Nullable } from '@audius/common/utils'
+import {
+  SelectablePill,
+  IconSpecialAccess,
+  IconCollectible,
+  IconSearch,
+  IconVisibilityHidden,
+  IconVisibilityPublic,
+  IconCart,
+  Paper,
+  Flex,
+  FilterButton,
+  TextInput,
+  TextInputSize
+} from '@audius/harmony'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { TracksTable, TracksTableColumn } from 'components/tracks-table'
+import { useGoToRoute } from 'hooks/useGoToRoute'
+
+import { getDashboardTracksStatus } from '../store/selectors'
+import { fetchTracks } from '../store/slice'
+const { getAccountAlbums } = accountSelectors
+
+// Pagination Constants
+export const tablePageSize = 50
+
+const messages = {
+  filterInputPlacehoder: 'Search Tracks',
+  allReleases: 'All Releases',
+  public: 'Public',
+  premium: 'Premium',
+  specialAcess: 'SpecialAccess',
+  gated: 'Gated',
+  hidden: 'Hidden',
+  tracks: 'Tracks',
+  albums: 'Albums',
+  search: (type: Pills) =>
+    `Search ${type === Pills.TRACKS ? 'Tracks' : 'Albums'}`
+}
+
+const tracksTableColumns: TracksTableColumn[] = [
+  'spacer',
+  'trackName',
+  'releaseDate',
+  'length',
+  'plays',
+  'reposts',
+  'overflowMenu'
+]
+
+const albumTableColumns: TracksTableColumn[] = [
+  'spacer',
+  'trackName',
+  'releaseDate',
+  'reposts',
+  'overflowMenu'
+]
+
+export type DataSourceTrack = Track & {
+  key: string
+  name: string
+  date: string
+  time?: number
+  saves: number
+  reposts: number
+  plays: number
+}
+
+export type DataSourceAlbum = Collection & {
+  key: string
+  name: string
+  date: string
+  saves: number
+  reposts: number
+}
+
+enum Pills {
+  TRACKS,
+  ALBUMS
+}
+
+enum TrackFilters {
+  PUBLIC = 'Public',
+  PREMIUM = 'Premium',
+  SPECIAL_ACCESS = 'SpecialAccess',
+  COLLECTIBLE_GATED = 'CollectibleGated',
+  HIDDEN = 'Hidden'
+}
+
+enum AlbumFilters {
+  PUBLIC = 'Public',
+  PREMIUM = 'Premium',
+  HIDDEN = 'Hidden'
+}
+
+const formatAlbum = (album: Collection): DataSourceAlbum => {
+  return {
+    ...album,
+    key: String(album.playlist_id),
+    name: album.playlist_name,
+    date: album.created_at,
+    saves: album.save_count,
+    reposts: album.repost_count
+  }
+}
+
+type ArtistContentSectionProps = {
+  tracks: DataSourceTrack[]
+  account: User
+}
+
+export const ArtistContentSection = ({
+  tracks,
+  account
+}: ArtistContentSectionProps) => {
+  const dispatch = useDispatch()
+  const goToRoute = useGoToRoute()
+  const [filterText, setFilterText] = useState('')
+  const [selectedPill, setSelectedPill] = useState(Pills.TRACKS)
+  const [selectedTrackFilter, setSelectedTrackFilter] =
+    useState<Nullable<TrackFilters>>(null)
+  const [selectedAlbumFilter, setSelectedAlbumFilter] =
+    useState<Nullable<AlbumFilters>>(null)
+  const isTracks = selectedPill === Pills.TRACKS
+  const tracksStatus = useSelector(getDashboardTracksStatus)
+
+  const albumsUnformatted = useSelector(getAccountAlbums) ?? []
```

Comment:
so that's true, but we also fetch all the account's albums in a saga on startup, and `getAccountAlbums` grabs the ids from the sparse albums and looks them up in the cache